how-to/customize-workspace/client/src/modules/auth/example/auth-provider.ts

```diff
@@ -0,0 +1,277 @@
+let logInfo = console.log;
+let logWarning = console.warn;
+let logError = console.error;
+let authRequiredCallback: (authenticationRequired: boolean) => void;
+let authenticated: boolean;
+let authOptions: ExampleOptions;
+interface ExampleOptions {
+	autoLogin: boolean;
+	authenticatedUrl: string;
+	loginUrl: string;
+	logoutUrl: string;
+	loginHeight: number;
+	loginWidth: number;
+	checkLoginStatusInSeconds: number;
+	checkSessionValidityInSeconds: number;
+}
+
+const EXAMPLE_AUTH_AUTHENTICATED_KEY = "EXAMPLE_AUTH_IS_AUTHENTICATED";
+
+async function openLoginWindow(url: string): Promise<OpenFin.Window> {
+	return fin.Window.create({
+		name: "example-auth-log-in",
+		alwaysOnTop: true,
+		maximizable: false,
+		minimizable: false,
+		autoShow: false,
+		defaultCentered: true,
+		defaultHeight: authOptions.loginHeight ?? 250,
+		defaultWidth: authOptions.loginWidth ?? 400,
+		includeInSnapshots: false,
+		resizable: false,
+		showTaskbarIcon: false,
+		saveWindowState: false,
+		url
+	});
+}
+
+async function openLogoutWindow(url: string): Promise<OpenFin.Window> {
+	return fin.Window.create({
+		name: "example-auth-log-out",
+		maximizable: false,
+		minimizable: false,
+		autoShow: false,
+		defaultCentered: true,
+		defaultHeight: authOptions.loginHeight ?? 250,
+		defaultWidth: authOptions.loginWidth ?? 400,
+		includeInSnapshots: false,
+		resizable: false,
+		showTaskbarIcon: false,
+		saveWindowState: false,
+		url
+	});
+}
+
+async function checkAuth(url: string): Promise<boolean> {
+	const windowToCheck = await fin.Window.create({
+		name: "example-auth-check-window",
+		alwaysOnTop: true,
+		maximizable: false,
+		minimizable: false,
+		autoShow: false,
+		defaultHeight: authOptions.loginHeight ?? 250,
+		defaultWidth: authOptions.loginWidth ?? 400,
+		includeInSnapshots: false,
+		resizable: false,
+		showTaskbarIcon: false,
+		saveWindowState: false,
+		url
+	});
+	let isAuthenticated = false;
+	try {
+		const info = await windowToCheck.getInfo();
+		if (info.url === authOptions.authenticatedUrl) {
+			isAuthenticated = true;
+		}
+	} catch (error) {
+		logError(`Error encountered while checking session.`, error);
+	} finally {
+		if (windowToCheck !== undefined) {
+			await windowToCheck.close(true);
+		}
+	}
+	return isAuthenticated;
+}
+
+async function getAuthenticationFromUser(): Promise<boolean> {
+	return new Promise<boolean>((resolve, reject) => {
```

Comment:
Refactor this remove `new Promise` and just use `const win = await openLoginWIndow`